dsmiddleware/rpcretry/rpcretry.go

```diff
@@ -72,12 +73,24 @@ func (rh *retryHandler) try(ctx context.Context, logPrefix string, f func() erro
 		} else if _, ok := err.(datastore.MultiError); ok {
 			// If MultiError returns, it should not be fixed even if it is retried
 			return
-		} else if err == context.DeadlineExceeded || err == context.Canceled {
+		} else if errors.Is(err, context.DeadlineExceeded) || errors.Is(err, context.Canceled) {
 			// for appengine datastore
 			return
-		} else if code := status.Code(err); code == codes.DeadlineExceeded || code == codes.Canceled {
+		} else if code := status.Code(err); code != codes.Unknown {
 			// for cloud datastore
-			return
+			// https://cloud.google.com/datastore/docs/concepts/errors?hl=en#error_codes
+			switch code {
+			case codes.Aborted,
```

Comment:
The Recommended Action `code=INTERNAL` says 
> Do not retry this request more than once.
Can it be implemented?
If not, I think it is better to comment it as TODO.

Reply:
thanks. address comment.